src/lib.rs

```diff
@@ -1,35 +1,39 @@
-// Declaring our library as `no-std` unconditionally lets us be consistent
-// in how we `use` items from `std` or `core`
-#![no_std]
+// Declare our library as `no-std` unless the user opts in for std
+// features.
```

Comment:
Personally I like to be unconditional with the `#![no_std]` attribute, which means you don't get the nice prelude when `std` is available, but does keep the library internals consistent when pulling in items that aren't in `libcore`'s prelude.

Reply:
I'm with you there, and will update no-std-compat to do the same.